solidity/random-beacon/contracts/DKGValidator.sol

```diff
@@ -0,0 +1,230 @@
+        // in ascending order.
+        if (
+            signingMembersIndices[0] < 1 ||
+            signingMembersIndices[signingMembersIndices.length - 1] > 64
+        ) {
+            return (false, "Corrupted signing member indices");
+        }
+        for (uint256 i = 1; i < signingMembersIndices.length; i++) {
+            if (signingMembersIndices[i - 1] >= signingMembersIndices[i]) {
+                return (false, "Corrupted signing member indices");
+            }
+            uint256 last = signingMembersIndices[
+                signingMembersIndices.length - 1
+            ];
+            if (last < 1 || last > groupSize) {
+                return (false, "Corrupted signing member indices");
+            }
+        }
+
+        return (true, "");
+    }
+
+    /// @notice Performs validation of group members as declared in DKG
+    ///         result against group members selected by the sortition pool.
+    /// @param seed seed used to start the DKG and select group members
+    /// @return true if group members matches; false otherwise
+    function validateGroupMembers(DKG.Result calldata result, uint256 seed)
+        public
+        view
+        returns (bool)
+    {
+        // Compute the actual group members hash by selecting actual members IDs
+        // based on seed used for current DKG execution.
+        bytes32 actualGroupMembersHash = keccak256(
+            abi.encodePacked(
+                sortitionPool.selectGroup(groupSize, bytes32(seed))
+            )
+        );
+
+        // TODO: check what is more efficient - computing hash or iterating
+        return
+            keccak256(abi.encodePacked(result.members)) ==
+            actualGroupMembersHash;
+    }
+
+    /// @notice Performs validation of signatures supplied in DKG result.
+    ///         Note that this function does not check if addresses which
+    ///         supplied signatures supporting the result are the ones selected
+    ///         to the group by sortition pool. This function should be used
+    ///         together with `validateGroupMembers`.
+    /// @param startBlock DKG start block
+    /// @return true if group members matches; false otherwise
+    function validateSignatures(DKG.Result calldata result, uint256 startBlock)
+        public
+        view
+        returns (bool)
+    {
+        bytes32 hash = keccak256(
+            abi.encodePacked(
+                result.groupPubKey,
+                result.misbehavedMembersIndices,
+                startBlock
+            )
+        );
+
+        address[] memory membersAddresses = sortitionPool.getIDOperators(
+            result.members
+        );
+
+        bytes memory current; // Current signature to be checked.
+
+        uint256 signaturesCount = result.signatures.length / 65;
+        for (uint256 i = 0; i < signaturesCount; i++) {
+            uint256 memberIndex = result.signingMembersIndices[i];
+
+            current = result.signatures.slice(65 * i, 65);
```

Comment:
If we decide to extract `signatureBytesSize` we can use it here.

Reply:
https://github.com/keep-network/keep-core/pull/2734/commits/5bc95364647baabb9b6c7951805b9fb2681dcdfb